### DISCUSSION TITLE
Error handling for parallel actions

### DISCUSSION THREAD

**Discussion by paarthdassani** [serverlessworkflow/specification#809] [Q&A] [answered]
I was confused as to how the runtime should behave wrt handling errors for parallel actions executed within a foreach state.

1. When executing an action parallelly for a inputCollection, we could get different types of errors on different executions of the actions. 
The error handling is specified at the state level onError definitions. How do we handle the case where these different errors have different transitions defined?
2. Is there a way to specify handling per execution in a parallel scenario? (Eg: I just want to append the details in a separate "failureDetails" list for the failed actions, and keep the workflow going for the rest?)
3. When executing parallel actions and one of them fails early, do we wait for the other executions to complete before the error is handled?

**Comment by cdavernas** [comment, marked as answer]
Error handling is being refactored as we speak, for its current state makes it unusable and forces runtime specific implementations.

I'd therefore recommend waiting for 0.9 before attempting implementing it. Same goes for retries and, possibly, compensation.

**Reply by paarthdassani** [reply]
Thanks for the info! Are there any discussions/resources about the approach being planned?
If there is already a plan, I could try to implement things in a way that makes it easier to adhere to the spec when 0.9 is out..

**Reply by cdavernas** [reply]
We are envisionning the ProblemDetails spec to describe errors. 
Handlers should look a lot like try/catch/except block, declarable and referencable throughout, and should mostly use status codes to describe errors, and JSON paths to describe sources.

Be welcome in next week's meeting if you have questions/ideas/suggestions.
